sklearn/tree/_tree.pyx

```diff
@@ -438,7 +438,7 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
         n_node_samples = end - start
         is_leaf = ((depth > self.max_depth) or
                    (n_node_samples < self.min_samples_split) or
-                   (n_node_samples < 2 * self.min_samples_leaf) or
+                   (n_node_samples < self.min_samples_leaf) or
```

Comment:
In fact isn't `weighted_n_node_samples < self.min_weight_leaf`opposite as well? We're declaring something a leaf if it has less than min_weight_leaf...which seems wrong to me?